croutine.c

```diff
@@ -378,5 +378,30 @@
 
         return xReturn;
     }
+/*-----------------------------------------------------------*/
+
+    #if ( configSUPPORT_REINITIALISE_INTERNAL_VARIABLES == 1 )
```

Comment:
Relevant for all of these functions. But instead of wrapping these functions inside of this config check, why not just include them and let the linker script exclude them if they aren't used?

Reply:
The intention here:
* vPortHeapReinitialiseVariables is not a mandatory heap port interface. Without this config, we will need another config to for this.
* Make it more clear that these re-initialize functions are only required when application needs to restart scheduler.

Reply:
The macro is removed as suggested. Let me resolve this conversation.